rcm_nexus/command.py

```diff
@@ -57,79 +83,88 @@ def push(repo, environment, product, version, ga=False, debug=False):
     More information: https://mojo.redhat.com/docs/DOC-1199638
     """
 
-    nexus_config = config.load(environment, debug=debug)
-
-    session = Session(nexus_config, debug=debug)
-    print("Pushing: %s content to: %s" % (repo.decode("utf-8"), environment))
-
-    try:
-        # produce a set of clean repository zips for PUT upload.
-        zips_dir = tempfile.mkdtemp()
-        print("Creating ZIP archives in: %s" % zips_dir)
-        if os.path.isdir(repo):
-            print("Processing repository directory: %s" % repo)
-
-            # Walk the directory tree, and create a zip.
-            zip_paths = archive.create_partitioned_zips_from_dir(repo, zips_dir)
-        else:
-            print("Processing repository zip archive: %s" % repo)
+    nexus_config = config.load(environment)
+    npm_archive_type = npm.detect_npm_archive(repo)
+    if npm_archive_type != npm.NpmArchiveType.NOT_NPM:
+        npm.push(nexus_config, repo, npm_archive_type, product, debug=debug)
+    else:
+        if nexus_config.get_profile_type(product) != config.ProfileType.JAVA:
+            print(product + " is not a java product!", file=sys.stderr)
+            exit(1)
+
+        session = Session(nexus_config, debug=debug)
+        print("Pushing: %s content to: %s" % (repo.decode("utf-8"), environment))
+
+        zips_dir = None
+        try:
+            # produce a set of clean repository zips for PUT upload.
+            zips_dir = tempfile.mkdtemp()
+            print("Creating ZIP archives in: %s" % zips_dir)
+            if os.path.isdir(repo):
+                print("Processing repository directory: %s" % repo)
+
+                # Walk the directory tree, and create a zip.
+                zip_paths = archive.create_partitioned_zips_from_dir(repo, zips_dir)
+            else:
+                print("Processing repository zip archive: %s" % repo)
 
-            # Open the zip, walk the entries and normalize the structure to
-            # clean zip
-            zip_paths = archive.create_partitioned_zips_from_zip(
-                repo, zips_dir, debug=debug
-            )
+                # Open the zip, walk the entries and normalize the structure to
+                # clean zip
+                zip_paths = archive.create_partitioned_zips_from_zip(
+                    repo, zips_dir, debug=debug
+                )
 
-        # Open new staging repository with description
-        staging_repo_id = staging.start_staging_repo(session, nexus_config, product, version, ga)
+            # Open new staging repository with description
+            staging_repo_id = staging.start_staging_repo(session, nexus_config, product, version, ga)
```

Comment:
Does NPM not use staging? Why is this only for non-NPM packages?

Reply:
No, the staging suite is only usable for java. There is no support of staging for npm in Nexus. It is direct deployment to npm repository.